### PR TITLE
Make skip of trigger form in UI if no params are defined configurable

### LINKED ISSUE #33344: Not able to trigger DAG with config from UI if param is not defined in a DAG and dag_run.conf is used
### Apache Airflow version

2.7.0rc1

### What happened

As per https://github.com/apache/airflow/pull/31583, now we can only run DAG with config from UI if DAG has params, however, if a DAG is using dag_run.conf there is no way to run with config from UI and as dag_run.conf is not deprecated most of the users will be impacted by this

@hussein-awala  also mentioned it in his [voting](https://lists.apache.org/thread/zd9ppxw1xwxsl66w0tyw1wch9flzb03w)

### What you think should happen instead

I think there should be a way to provide param values from UI when dag_run.conf is used in DAG

### How to reproduce

Use below DAG in 2.7.0rc and you will notice there is no way to provide conf value from airflow UI

DAG CODE

```
from airflow.models import DAG
from airflow.operators.bash import BashOperator
from airflow.operators.python import PythonOperator
from airflow.utils.dates import days_ago

dag = DAG(
dag_id="trigger_target_dag",
default_args={"start_date": days_ago(2), "owner": "Airflow"},
tags=["core"],
schedule_interval=None, # This must be none so it's triggered by the controller
is_paused_upon_creation=False, # This must be set so other workers can pick this dag up. mabye it's a bug idk
)

def run_this_func(**context):
print(
f"Remotely received value of {context['dag_run'].conf['message']} for key=message "
)

run_this = PythonOperator(
task_id="run_this",
python_callable=run_this_func,
dag=dag,
)

bash_task = BashOperator(
task_id="bash_task",
bash_command='echo "Here is the message: $message"',
env={"message": '{{ dag_run.conf["message"] if dag_run else "" }}'},
dag=dag,
)
```

### Operating System

MAS os Monterey

### Versions of Apache Airflow Providers

_No response_

### Deployment

Astronomer

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jscheffl**
This PR is a hotfix for Airflow 2.7.0rc1 adding a configuration option so that the trigger form on UI can be displayed also for DAGs w/o `params` defined which was removed in #31583.

How to test?
- Start Airflow Webserver on this PR
- Check that form still displays (always) when triggering DAG "example_bash_operator" - as this has a param defined
- Trigger the example_python_operator and see it starts w/o displaying UI
- Stop airflow webserver and start it with `AIRFLOW__WEBSERVER__SKIP_TRIGGER_FORM_IF_NO_PARAM=False airflow webserver -d` and trigger the example_python_operator - See it displays the JSON entry form (again)

Screenshot 2023-08-12 at 18-59-47 Airflow

closes: #33344

### CODE REVIEW

**Comment by jens-scheffler-bosch** [comment]
To be merged based on vote in https://lists.apache.org/thread/pvvn2hxf03nrjm3l9crhmzw5yls6ftpn

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -1779,6 +1779,15 @@ webserver:
       type: string
       example: "sha256"
       default: "md5"
+    skip_trigger_form_if_no_params:
```
Comment (pierrejeambrun) [inline airflow/config_templates/config.yml]:
```suggestion
    show_trigger_form_if_no_params:
```
Will avoid a double negation in the code such as `not conf.getboolean("webserver", "skip_trigger_form_if_no_params")`. A bit easier to read IMO

File: airflow/www/views.py [hunk 2]
```diff
@@ -2031,6 +2031,7 @@ def trigger(self, dag_id: str, session: Session = NEW_SESSION):
                     form_fields[k]["schema"]["custom_html_form"]
                 )
         ui_fields_defined = any("const" not in f["schema"] for f in form_fields.values())
+        show_form_if_no_params = not conf.getboolean("webserver", "skip_trigger_form_if_no_params")
```
Comment (pierrejeambrun) [inline airflow/www/views.py]:
```suggestion
        show_trigger_form_if_no_params = conf.getboolean("webserver", "show_trigger_form_if_no_params")
```

**Comment by pierrejeambrun** [comment]
cc: @ephraimbuddy Can be grouped with `Trigger Button - Implement Part 2 of AIP-50 (#31583)` in the release note I believe.

File: airflow/www/views.py [hunk 3]
```diff
@@ -2031,6 +2031,7 @@ def trigger(self, dag_id: str, session: Session = NEW_SESSION):
                     form_fields[k]["schema"]["custom_html_form"]
                 )
         ui_fields_defined = any("const" not in f["schema"] for f in form_fields.values())
+        show_form_if_no_params = conf.getboolean("webserver", "show_trigger_form_if_no_params")
```
Comment (pierrejeambrun) [inline airflow/www/views.py]:
```suggestion
        show_trigger_form_if_no_params = conf.getboolean("webserver", "show_trigger_form_if_no_params")
```

File: airflow/www/views.py [hunk 4]
```diff
@@ -2057,7 +2058,7 @@ def trigger(self, dag_id: str, session: Session = NEW_SESSION):
             if isinstance(run_conf, dict) and any(run_conf)
         }
 
-        if request.method == "GET" and ui_fields_defined:
+        if request.method == "GET" and (ui_fields_defined or show_form_if_no_params):
```
Comment (pierrejeambrun) [inline airflow/www/views.py]:
```suggestion
        if request.method == "GET" and (ui_fields_defined or show_trigger_form_if_no_params):
```

**Review by pierrejeambrun — APPROVED** [review]
more nit :)

**Review by hussein-awala — APPROVED** [review]

File: airflow/www/views.py (hunk 3 above)
Comment (jscheffl) [reply airflow/www/views.py]: ...and I thought I can save a few bytes of typing - as we are in trigger form anyway :-D don't need to repeat the term trigger :-D

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Approved, but I think we need a newsfragment, describing essentially:

If you used to use dagrun config dictionary of configurations when you run the DAG, you shoud either:

* enable the new param ("show_trigger_form_if_no_params") to bring back old behaviour
* add params to your DAG as described in ...

Then I am super-happy

**Comment by ephraimbuddy** [comment]
> Approved, but I think we need a newsfragment, describing essentially:
> 
> If you used to use dagrun config dictionary of configurations when you run the DAG, you shoud either:
> 
> * enable the new param ("show_trigger_form_if_no_params") to bring back old behaviour
> * add params to your DAG as described in ...
> 
> Then I am super-happy

Yeah, agree with you

**Review by ephraimbuddy — APPROVED** [review]

**Comment by ephraimbuddy** [comment]
> To be merged based on vote in https://lists.apache.org/thread/pvvn2hxf03nrjm3l9crhmzw5yls6ftpn

Merged since the majority are voting in favour of it

**Pull request merged**